catroid/src/main/java/org/catrobat/catroid/content/Look.java

```diff
@@ -377,6 +368,17 @@ public float getDirectionInUserInterfaceDimensionUnit() {
 
 	public void setRotationMode(int mode) {
 		rotationMode = mode;
+		rotateLookDataIfNeeded(mode);
+	}
+
+	private void rotateLookDataIfNeeded(int mode) {
+		if (mode != Look.ROTATION_STYLE_LEFT_RIGHT_ONLY && sprite.look.isFlipped()) {
+			getLookData().getTextureRegion().flip(true, false);
+		}
+		boolean orientedLeft = sprite.look.getDirectionInUserInterfaceDimensionUnit() < 0;
+		if (mode == Look.ROTATION_STYLE_LEFT_RIGHT_ONLY && orientedLeft) {
+			getLookData().getTextureRegion().flip(true, false);
+		}
```

Comment:
ah, i see, thanks for clarification :)
also, it is more readable with your last change ;)